examples/null_mapping.q

```diff
@@ -6,47 +6,187 @@
+lstr_data:N?("start";"stop";"alert";"acknowledge";"");
+lstr_data[1]:"stop"
+bin_data:N?("x"$"start";"x"$"stop";"x"$"alert";"x"$"acknowledge";"x"$"");
+bin_data[2]:"x"$"alert"
+lbin_data:N?("x"$"start";"x"$"stop";"x"$"alert";"x"$"acknowledge";"x"$"");
+lbin_data[3]:"x"$"acknowledge"
+fbin_data:N?0Ng;
+fbin_data[4]:0Ng;
+
+d32_data:N?(2006.07.21;2008.07.18;2012.07.16;2014.07.15;2016.07.11);
+d32_data[0]:2006.07.21;
+d64_data:N?(2015.01.01D00:00:00.000000000;2017.01.01D00:00:00.000000000;2018.01.01D00:00:00.000000000;2019.01.01D00:00:00.000000000;2020.01.01D00:00:00.000000000);
+d64_data[1]:2015.01.01D00:00:00.000000000;
+tstamp_data:N?(2015.01.01D00:00:00.000000000;2014.01.01D00:00:00.000000000;2013.01.01D00:00:00.000000000;2012.01.01D00:00:00.000000000;2011.01.01D00:00:00.000000000);
+tstamp_data[2]:2011.01.01D00:00:00.000000000;
+t64_data:N?(12:00:00.000000000;13:00:00.000000000;14:00:00.000000000;15:00:00.000000000;16:00:00.000000000);
+t64_data[3]:12:00:00.000000000;
+dur_data:N?(12:00:00.000000000;13:00:00.000000000;14:00:00.000000000;15:00:00.000000000;16:00:00.000000000);
+dur_data[4]:12:00:00.000000000;
+
+f16_data:N?100h;
+f16_data[0]:9h;
+t32_data:N?(09:01:02.042;08:01:02.042;07:01:02.042;06:01:02.042;05:01:02.042);
+t32_data[1]:09:01:02.042;
+mint_data:N?(2006.07m;2006.06m;2006.05m;2006.04m;2006.03m);
+mint_data[2]:2006.07m;
+dtint_data:N?(12:00:00.000000000;11:00:00.000000000;10:00:00.000000000;09:00:00.000000000;08:00:00.000000000);
+dtint_data[3]:12:00:00.000000000;
+
+// Combine the data for all columns
+short_data:(ts_data;bool_data;ui8_data;i8_data;ui16_data;i16_data);
+long_data:(ts_data;ui32_data;i32_data;ui64_data;i64_data);
+float_data:(ts_data;f32_data;f64_data;dec_data);
+str_data:(ts_data;str_data;lstr_data;bin_data;lbin_data;fbin_data);
+time_data:(ts_data;d32_data;d64_data;tstamp_data;t64_data;dur_data)
+other_data:(ts_data;f16_data;t32_data;mint_data;dtint_data)
 
 // Pretty print the Arrow table populated from the array data
-.arrowkdb.tb.prettyPrintTable[schema;array_data;options]
+options[`DECIMAL128_AS_DOUBLE]:1
+.arrowkdb.tb.prettyPrintTable[short_schema;short_data;options]
+.arrowkdb.tb.prettyPrintTable[long_schema;long_data;options]
+.arrowkdb.tb.prettyPrintTable[float_schema;float_data;options]
+.arrowkdb.tb.prettyPrintTable[str_schema;str_data;options]
+.arrowkdb.tb.prettyPrintTable[time_schema;time_data;options]
+.arrowkdb.tb.prettyPrintTable[other_schema;other_data;options]
 
 options[`PARQUET_VERSION]:`V2.0
-.arrowkdb.pq.writeParquet["null_mapping.parquet";schema;array_data;options]
+.arrowkdb.pq.writeParquet["null_mapping_short.parquet";short_schema;short_data;options]
+.arrowkdb.pq.writeParquet["null_mapping_long.parquet";long_schema;long_data;options]
+.arrowkdb.pq.writeParquet["null_mapping_float.parquet";float_schema;float_data;options]
+.arrowkdb.pq.writeParquet["null_mapping_str.parquet";str_schema;str_data;options]
+.arrowkdb.pq.writeParquet["null_mapping_time.parquet";time_schema;time_data;options]
```

Comment:
I'd also read the data back and compare - the null mapping should work in both directions.

Reply:
(once you add read support)

Reply:
Since null mapping is not yet ready for backward direction, I can only compare reloaded parquet files with that arrow data, can't I?